type.go

```diff
@@ -347,6 +350,46 @@ func (t booleanType) ConvertValue(val Value, typ Type) (Value, error) {
 	}
 }
 
+// converting v to int64 guarantees that we will not lose data
+func assignToIntBasedPointer(dst reflect.Value, v int64) {
+	underlying := reflect.TypeOf(dst).Elem().Kind()
+	switch underlying {
+	case reflect.Int8:
+		val := int8(v)
+		dst.Set(reflect.ValueOf(&val))
+	case reflect.Int16:
+		val := int16(v)
+		dst.Set(reflect.ValueOf(&val))
+	case reflect.Int32:
+		val := int32(v)
+		dst.Set(reflect.ValueOf(&val))
+	case reflect.Int64:
+		val := int64(v)
+		dst.Set(reflect.ValueOf(&val))
+	case reflect.Uint8:
+		val := uint8(v)
+		dst.Set(reflect.ValueOf(&val))
+	case reflect.Uint16:
+		val := uint16(v)
+		dst.Set(reflect.ValueOf(&val))
+	case reflect.Uint32:
+		val := uint32(v)
+		dst.Set(reflect.ValueOf(&val))
+	case reflect.Uint64:
+		val := uint64(v)
+		dst.Set(reflect.ValueOf(&val))
+	case reflect.Int:
+		val := int(v)
+		dst.Set(reflect.ValueOf(&val))
+	case reflect.Uint:
+		val := uint(v)
+		dst.Set(reflect.ValueOf(&val))
+	default:
+		// trigger a panic for type correction
+		dst.Set(reflect.ValueOf(v))
+	}
```

Comment:
I think we can simplify this further by merging it with the `AssignValue` with something like this:
```go
func assignInt(dst reflect.Value, v uint64) {
  switch dst.Kind() {
  ...
  case reflect.Pointer:
    ptr := reflect.New(dst.Type())
    dst.Set(ptr)
    assignInt(ptr.Elem(), v)
  ...
  }
}
```

Reply:
you are right, I am a bit new to reflect, so I didn't come up with that solution, my bad.